### DISCUSSION TITLE
Is there a way to add a button or something that could collapse/uncollapse all the admonitions in a page?

### DISCUSSION THREAD

**Discussion by jonaprieto** [squidfunk/mkdocs-material#7685] [Q&A] [answered]
As the title says.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @jonaprieto,
yes, it's possible to do so using extra JavaScript:
- https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript

You can reference the other discussions, which also were related to JS:
- https://github.com/squidfunk/mkdocs-material/discussions/6849
- https://github.com/squidfunk/mkdocs-material/discussions/6512#discussioncomment-10640209

The logic should be simple, keep track of the state of the button to decide whether to collapse or uncollapse, after press go over all `details` tags on the page and change the state from the button, since you're enforcing the state from the button you don't need to check the state of the separate `details`.

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#open

If you need more guidance, please follow the https://dontasktoask.com/ idea and be specific about where you're having issues.
For the basics of JavaScript there are also other resources online, perhaps more fit to help you out than this discussion board :v:

**Reply by jonaprieto** [reply]
Thanks, the links were useful indeed.
